presto-main/src/test/java/com/facebook/presto/block/TestInterleavedBlock.java

```diff
@@ -0,0 +1,95 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.block;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.block.InterleavedBlockBuilder;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+
+public class TestInterleavedBlock
+        extends AbstractTestBlock
+{
+    private static final int[] SIZE_0 = new int[] {16, 0, 13, 1, 2, 11, 4, 7};
+    private static final int SIZE_1 = 8;
+    private static final int POSITION_COUNT = SIZE_0.length;
+    private static final int COLUMN_COUNT = 2;
+
+    @Test
+    public void test()
+    {
+        Slice[] expectedValues = createExpectedValues();
+        assertValues(expectedValues);
+        assertValues((Slice[]) alternatingNullValues(expectedValues));
+    }
+
+    private void assertValues(Slice[] expectedValues)
+    {
+        InterleavedBlockBuilder blockBuilder = new InterleavedBlockBuilder(ImmutableList.of(VARCHAR, BIGINT), new BlockBuilderStatus(), 10);
+
+        for (Slice expectedValue : expectedValues) {
+            if (expectedValue == null) {
+                blockBuilder.appendNull();
+            }
+            else {
+                blockBuilder.writeBytes(expectedValue, 0, expectedValue.length()).closeEntry();
+            }
+        }
+        assertBlock(blockBuilder, expectedValues);
+        assertBlock(blockBuilder.build(), expectedValues);
+    }
+
+    @Override
+    protected void assertBlockPosition(Block block, int position, Slice expectedValue)
+    {
+        assertPositionValue(block, position, expectedValue);
+        assertPositionValue(block.getSingleValueBlock(position), 0, expectedValue);
+        assertPositionValue(block.getRegion(roundDown(position, COLUMN_COUNT), COLUMN_COUNT), position % COLUMN_COUNT, expectedValue);
+        assertPositionValue(block.getRegion(0, roundUp(position + 1, COLUMN_COUNT)), position, expectedValue);
+        assertPositionValue(block.getRegion(roundDown(position, COLUMN_COUNT), block.getPositionCount() - roundDown(position, COLUMN_COUNT)), position % COLUMN_COUNT, expectedValue);
+        assertPositionValue(block.copyRegion(roundDown(position, COLUMN_COUNT), COLUMN_COUNT), position % COLUMN_COUNT, expectedValue);
+        assertPositionValue(block.copyRegion(0, roundUp(position + 1, 2)), position, expectedValue);
+        assertPositionValue(block.copyRegion(roundDown(position, COLUMN_COUNT), block.getPositionCount() - roundDown(position, COLUMN_COUNT)), position % COLUMN_COUNT, expectedValue);
+    }
+
+    private int roundDown(int position, int divisor)
+    {
+        return position / divisor * divisor;
+    }
+
+    private int roundUp(int position, int divisor)
+    {
+        if (roundDown(position, divisor) == position) {
```

Comment:
you could simplify this `if` to `return roundDown(position + divisor - 1, divisor)`